openwisp_controller/config/api/serializers.py

```diff
@@ -50,7 +50,7 @@ def validate_vpn(self, value):
         Ensure that VPN can't be added when
         template `Type` is set to `Generic`.
         """
-        if self.context['request'].data['type'] == 'generic':
+        if self.initial_data.get('type') == 'generic' and value is not None:
```

Comment:
@ManishShah120 when you fix issues, try to always write basic tests, make it a habit, it's a good one